plugins/node/opentelemetry-instrumentation-pino/test/pino.test.ts

```diff
@@ -87,10 +106,16 @@ describe('PinoInstrumentation', () => {
     }
   }
 
-  before(() => {
-    instrumentation = new PinoInstrumentation();
+  function setup() {
+    instrumentation = new PinoInstrumentation({
+      logKeys,
+    });
     instrumentation.enable();
     pino = require('pino');
+  }
+
+  before(() => {
+    setup();
```

Comment:
If I understand this correct this causes that all tests below use the new keys, not just the new `injects span context to records with custom keys` test.

Reply:
That's right. I was having trouble configuring just one test independently of the others with the current setup. I can spend a bit more time on it

Reply:
Cleaned them up but the tests are interfering with each other causing the "disabled instrumentation" block of tests to fail. If I comment out all the other `describe` blocks then they pass. I'm stumped at the moment but will revisit this in the next couple days

Reply:
Seems like the pino tests were written in a way to instrument just once and then run all tests.
Don't know if there was a reason. Theoretically `.disable()` should cause to de-instrument and therefore should allow side effect free tests.
But there might be cases where de-instrument doesn't work, e.g. if pino caches some stuff internal.

Reply:
I definitely don't have a perfect understanding of this patching, I tried creating a new instrumentation + explicitly calling `enable()` on it as well as clearing the require cache via `delete require.cache[require.resolve('pino')]` but it made no difference. Would love to hear alternatives to try as welll

Reply:
I broke the enabled + disabled tests out into their own modules to skirt the problem. Let me know what you think